iceoryx_hoofs/CMakeLists.txt

```diff
@@ -64,173 +64,32 @@ set(${PROJECT_NAME}_DIR ${CMAKE_CURRENT_LIST_DIR}/cmake
     FORCE
 )
 
-#
-########## set util internal target, needed by tests in components ##########
-#
-# TODO: we need something like internal public for internal build dependencies
-# instead of making the whole source folder public for internal dependency resolution
-
 if(BUILD_TEST AND NOT GTest_FOUND)
     find_package(GTest CONFIG REQUIRED)
 endif(BUILD_TEST AND NOT GTest_FOUND)
 
 if(GTest_FOUND) # only GTest_FOUND, just in case someone want's to use iceoryx_hoofs_testing without also building the tests
-
     setup_package_name_and_create_files(
         NAME ${PROJECT_NAME}_testing
         NAMESPACE iceoryx_hoofs_testing
         PROJECT_PREFIX ${PREFIX}
     )
 
-    add_library(iceoryx_hoofs_testing
-        STATIC
-        testing/mocks/time_mock.cpp
-        testing/mocks/error_handler_mock.cpp
-        testing/timing_test.cpp
-        testing/compile_test.cpp
-    )
-
-    add_library(iceoryx_hoofs_testing::iceoryx_hoofs_testing ALIAS iceoryx_hoofs_testing)
-
-    set_target_properties(iceoryx_hoofs_testing PROPERTIES VERSION ${PROJECT_VERSION})
-    if(PERFORM_CLANG_TIDY)
-        set_target_properties(
-            iceoryx_hoofs_testing PROPERTIES CXX_CLANG_TIDY "${PERFORM_CLANG_TIDY}"
-        )
-    endif(PERFORM_CLANG_TIDY)
-    set_target_properties(iceoryx_hoofs_testing PROPERTIES
-        CXX_STANDARD_REQUIRED ON
-        CXX_STANDARD ${ICEORYX_CXX_STANDARD}
-        POSITION_INDEPENDENT_CODE ON
-    )
-
-    if(TEST_WITH_ADDITIONAL_USER)
-        target_compile_definitions(iceoryx_hoofs_testing PUBLIC -DTEST_WITH_ADDITIONAL_USER)
-    endif(TEST_WITH_ADDITIONAL_USER)
-
-    target_include_directories(iceoryx_hoofs_testing
-        PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/testing/include>
-        $<INSTALL_INTERFACE:include/${PREFIX}>
-    )
-
-    target_link_libraries(iceoryx_hoofs_testing
-        PUBLIC
-            GTest::gtest
-            GTest::gmock
-        PRIVATE
-            ${CODE_COVERAGE_LIBS}
-            iceoryx_hoofs
-            ${CMAKE_DL_LIBS}
-    )
-
-    if(LINUX)
-        target_link_libraries(iceoryx_hoofs_testing PRIVATE rt)
-    endif()
-
-    target_compile_options(iceoryx_hoofs_testing PRIVATE ${ICEORYX_WARNINGS})
-
-    setup_install_directories_and_export_package(
-        TARGETS iceoryx_hoofs_testing
-        INCLUDE_DIRECTORY testing/include/
-    )
-
-    #
-    ########## find_package in source tree ##########
-    #
-    set(${PROJECT_NAME}_testing_DIR ${CMAKE_CURRENT_LIST_DIR}/cmake
-    CACHE FILEPATH
-    "${PROJECT_NAME}_testingConfig.cmake to make find_package(${PROJECT_NAME}_testing) work in source tree!"
-    FORCE
-    )
-
+    add_subdirectory(testing)
 endif(GTest_FOUND)
 
+#
+########## build iceoryx hoofs lib ##########
+#
+
 setup_package_name_and_create_files(
     NAME ${PROJECT_NAME}
     NAMESPACE iceoryx_hoofs
     PROJECT_PREFIX ${PREFIX}
 )
 
-#
-########## build iceoryx util lib ##########
-#
-
-add_library(iceoryx_hoofs
```

Comment:
Could you please revert this?

Since the header for this target are in the `include` folder which lies alongside the `source` folder, one would need to step out of `source` to add the `target_include_directories`.

For a future refactoring one could think of moving the include folder to source, like
```
iceoryx_hoofs
|- cmake
|- platform
|- source
|   |- include
|   |    |- iceoryx_hoofs
|   |- *.cpp or a cpp folder?
|- test
|- testing
```

@dkroenke what do you think?

Reply:
@elBoberido 
It looks a bit strange when there is an `include` folder below `source` and looks a bit unintuitive.
A `cpp` folder is strange too because the include folder should be then named `hpp` to be consistent...

I understand that the CMakeLists.txt in `iceoryx_hoofs` is quite large because of certain things like iceoryx_hoofs_testing or the Clang-Tidy statements.
Nevertheless it makes more sense imho to have in the `iceoryx_hoofs` folder a CMakeLists.txt that builds the iceoryx_hoofs lib itself. Other things like the testing lib, Gtest, Clang-Tidy or the packaging commands can be outsourced in external cmake files.
With that, the size of the CMakeLists.txt file in iceoryx_hoofs will be maintainable from my point of view.

Reply:
@dkroenke yes, this does not need to be decided now. I just thought about something like this https://palikar.github.io/posts/cmake_structure/

At some point we might want to modularize the hoofs a bit more, e.g. having `posix_wrapper` and `cxx` as separate modules. Instead of having a top-level `source` folder maybe we could have a `code` or `modules` folder, e.g.
```
iceoryx_hoofs
|- cmake
|- platform
|- modules
|   |- cxx
|   |   |- include
|   |   |- source
|   |- posix_wrapper
|   |    |- ...
|- test
|- testing
```

But like already mentioned, this is something which can be done in a later PR. At first we need to find out which direction we want to go. The current separation is already an improvement from what is in master.